### ISSUE TITLE
[Windows] v2.0.0 Sync Journal still Locked when Re-configuring

### ISSUE THREAD

**Issue by dragotin** [owncloud/client#3421]
On Windows: To reconfigure the client to a new server, the existing configuration (that had an connection error) was removed. Immediately the new connection wizard comes up, a new server was entered and the same local directory as before was picked.

However, the new sync could not be initialized because the sync journal was still locked. The client was coming up with a message box telling that. 

It was checked with LockHunter, it was the ownCloud Client itself that hold the lock.

**Comment by guruz** [comment]
My guess would be this is from the SocketApi?

**Comment by dragotin** [comment]
maybe also because a sync in the background is not yet terminated and keeps the thing open?

**Comment by guruz** [comment]
@ogoffart Btw, you made sure that ```alias``` used in ```socketapi.cpp``` is always unique, right?
By having aliases created elsewhere being unique even if they're for different accounts?

**Comment by guruz** [comment]
At least here on OS X I see the socketApi closing the DB, so I guess @dragotin 's theory makes more sense.

@dragotin Do you remember what you exactly did? Was it 1.8 with "Modify Account"? Since you now have to delete an account first I think this can be seen as fixed?

**Comment by ogoffart** [comment]
@guruz yes, aliases are unique,  but long tem aliases should disapear and Folder* should be used.

**Comment by phil-davis** [comment]
I got this a few hours ago, and did it again now. Having an account connection that did not display the "choose files to sync", I deleted the account from the multi-account client. Prior to deleting it said it was all synced. I let it leave the files in place.
Then I did Add an Account and fed in the same server URL and username+password and location for the client-side files as previously. It all looks good until this"Could not reset state" error dialog:
![owncloudsyncjournal](https://cloud.githubusercontent.com/assets/1535615/9009472/283b3c6c-37c1-11e5-8d70-b2313d5e0e72.png)
After doing Retry a few times, I gave up and pressed abort. Then the client looks like:
![owncloud-after-delete-add-account](https://cloud.githubusercontent.com/assets/1535615/9009558/cdc611ca-37c1-11e5-887c-ba9706e1d09d.png)
To get success I then delete the account, quit ownCloud, start ownCloud again and add the account.
Note: The first time my laptop was Windows7, I upgraded to Windows10 this afternoon and symptoms are all the same.
ownCloud client for Windows 2.0.0-nightly 20150731 (build 5317)

**Comment by guruz** [comment]
Thanks for testing. So this also leads to bogus UI state..

**Comment by guruz** [comment]
L

**Comment by guruz** [comment]
Looks like github also closes issues if the issue contains "close #issue"

**Comment by ogoffart** [comment]
Commit 2cf2935 seems to have fixed this issue.

**Comment by phil-davis** [comment]
Using 2.0.0.5367 on a client connected to just 1 account. I deleted the account. In only 1 second or so the dialog popped up to connect to an account (somebody recently reported something about a delay there of around 15 seconds - so maybe that go fixed yesterday also, I digress). I entered the same server and account information and local folder. I let it do the default "sync everything". It connected without problem (no sync journal locked stuff).
It downloaded just 1 folder that I had previously set to be not synced - good stuff, I had told it to now sync everything and it did exactly that with minimal sync, recognizing that most of the local files were already there and up-to-date.
Seems fixed to me.
